Erroring showing `System` after forecasts added via `transform_single_time_series!`

I'm trying to update some private code from PowerSystems.jl v0.28 to v1

Below is using PowerSystems v1.3.0 and InfrastructureSystems v1.3.6

I'm testing with a 3 bus system, which should have `SingleTimeSeries` for each of the `PowerLoad` and `ThermalStandard` 
```julia
julia> sys
System
======
System Units Base: SYSTEM_BASE
Base Power: 100.0
Base Frequency: 60.0

Components
==========
Num components: 26

8×3 DataFrame
 Row │ ConcreteType                   SuperTypes                         Count
     │ String                         String                             Int64
─────┼─────────────────────────────────────────────────────────────────────────
   1 │ Arc                            Topology <: Component <: Infrast…      3
   2 │ Bus                            Topology <: Component <: Infrast…      3
   3 │ Line                           ACBranch <: Branch <: Device <: …      3
   4 │ PhaseShiftingTransformer       ACBranch <: Branch <: Device <: …      1
   5 │ PowerLoad                      StaticLoad <: ElectricLoad <: St…      2
   6 │ StaticReserveGroup{ReserveUp}  Service <: Component <: Infrastr…      3
   7 │ ThermalStandard                ThermalGen <: Generator <: Stati…      2
   8 │ VariableReserve{ReserveUp}     Reserve{ReserveUp} <: Service <:…      9

TimeSeriesContainer
===================
Components with time series data: 4
Total StaticTimeSeries: 4
Total Forecasts: 0
Resolution: 60 minutes
```
I convert these time-series to forecasts, with `transform_single_time_series!`, and that _seems_ to work (no error or other output is shown)
```julia
julia> transform_single_time_series!(sys, 24, Hour(1))
```
but when I then see an error upon trying to show the resulting system:
```julia
julia> sys
System
======
System Units Base: SYSTEM_BASE
Base Power: 100.0
Base Frequency: 60.0

Components
==========
Num components: 26

8×3 DataFrame
 Row │ ConcreteType                   SuperTypes                         Count
     │ String                         String                             Int64
─────┼─────────────────────────────────────────────────────────────────────────
   1 │ Arc                            Topology <: Component <: Infrast…      3
   2 │ Bus                            Topology <: Component <: Infrast…      3
   3 │ Line                           ACBranch <: Branch <: Device <: …      3
   4 │ PhaseShiftingTransformer       ACBranch <: Branch <: Device <: …      1
   5 │ PowerLoad                      StaticLoad <: ElectricLoad <: St…      2
   6 │ StaticReserveGroup{ReserveUp}  Service <: Component <: Infrastr…      3
   7 │ ThermalStandard                ThermalGen <: Generator <: Stati…      2
   8 │ VariableReserve{ReserveUp}     Reserve{ReserveUp} <: Service <:…      9

TimeSeriesContainer
===================
Components with time series data: 4
Total StaticTimeSeries: 4
Total Forecasts: 4
Resolution: 60 minutes
Error showing value of type System:
ERROR: AssertionError: 1 hour == 0 seconds
Stacktrace:
  [1] macro expansion
    @ ~/.julia/packages/InfrastructureSystems/v75Hd/src/utils/assert_op.jl:34 [inlined]
  [2] get_initial_times(initial_timestamp::DateTime, count::Int64, interval::Hour)
    @ InfrastructureSystems ~/.julia/packages/InfrastructureSystems/v75Hd/src/utils/utils.jl:404
  [3] get_forecast_initial_times(params::InfrastructureSystems.ForecastParameters)
    @ InfrastructureSystems ~/.julia/packages/InfrastructureSystems/v75Hd/src/time_series_parameters.jl:67
  [4] get_forecast_initial_times
    @ ~/.julia/packages/InfrastructureSystems/v75Hd/src/time_series_parameters.jl:232 [inlined]
  [5] get_forecast_initial_times
    @ ~/.julia/packages/InfrastructureSystems/v75Hd/src/system_data.jl:601 [inlined]
  [6] show(io::IOContext{Base.TTY}, #unused#::MIME{Symbol("text/plain")}, data::InfrastructureSystems.SystemData)
    @ InfrastructureSystems ~/.julia/packages/InfrastructureSystems/v75Hd/src/utils/print.jl:93
  [7] show(io::IOContext{Base.TTY}, #unused#::MIME{Symbol("text/plain")}, sys::System)
    @ PowerSystems ~/.julia/packages/PowerSystems/yTZiA/src/utils/print.jl:18
  [8] (::REPL.var"#38#39"{REPL.REPLDisplay{REPL.LineEditREPL}, MIME{Symbol("text/plain")}, Base.RefValue{Any}})(io::Any)
```
Seemingly from an `AssertionError` in `get_forecast_initial_times(sys)`

--- a/src/component.jl
+++ b/src/component.jl
@@ -52,7 +52,6 @@ function _get_columns(start_time, count, ts_metadata::ForecastMetadata)
     if window_count > 1
         index = Int(offset / interval) + 1
     else
-        @assert_op interval == Dates.Millisecond(0)
         index = 1
     end
     if count === nothing

--- a/src/utils/utils.jl
+++ b/src/utils/utils.jl
@@ -400,11 +400,10 @@ function get_initial_times(
 )
     if count == 0
         return []
-    elseif count == 1
-        @assert_op interval == Dates.Second(0)
-        return range(initial_timestamp; stop = initial_timestamp, step = Dates.Second(1))
+    elseif interval == Dates.Second(0)
+        return [initial_timestamp]
     end
-    @assert interval != Dates.Second(0) "initial_timestamp=$initial_timestamp interval=$interval count=$count"
+
     return range(initial_timestamp; length = count, step = interval)
 end
 

--- a/test/test_time_series.jl
+++ b/test/test_time_series.jl
@@ -539,6 +539,35 @@ end
     @test IS.get_interval(forecast) == Dates.Second(0)
 end
 
+@testset "Test DeterministicSingleTimeSeries with interval = resolution" begin
+    sys = IS.SystemData(time_series_in_memory = true)
+    component = IS.TestComponent("Component1", 5)
+    IS.add_component!(sys, component)
+
+    resolution = Dates.Hour(1)
+    horizon = 24
+    dates = collect(
+        range(Dates.DateTime("2020-01-01T00:00:00"); length = horizon, step = resolution),
+    )
+    data = collect(1:horizon)
+    ta = TimeSeries.TimeArray(dates, data, [IS.get_name(component)])
+    name = "val"
+    ts = IS.SingleTimeSeries(name, ta)
+    IS.add_time_series!(sys, component, ts)
+
+    interval = resolution
+    IS.transform_single_time_series!(
+        sys,
+        IS.DeterministicSingleTimeSeries,
+        horizon,
+        interval,
+    )
+
+    initial_times = IS.get_forecast_initial_times(sys)
+    forecast = IS.get_time_series(IS.DeterministicSingleTimeSeries, component, name)
+    @test IS.get_interval(forecast) == interval
+end
+
 @testset "Test component removal with DeterministicSingleTimeSeries" begin
     sys = IS.SystemData()
     component = IS.TestComponent("Component1", 5)
@@ -1213,7 +1242,7 @@ end
     component = IS.TestComponent(name, 5)
     IS.add_component!(sys, component)
 
-    @test IS.get_forecast_initial_times(sys) == []
+    @test isempty(IS.get_forecast_initial_times(sys))
 
     resolution = Dates.Hour(1)
     initial_time = Dates.DateTime("2020-09-01")

--- a/test/test_time_series.jl
+++ b/test/test_time_series.jl
@@ -535,6 +535,8 @@ end
         interval,
     )
 
+    initial_times = collect(IS.get_forecast_initial_times(sys))
+    @test initial_times == [Dates.DateTime("2020-01-01T00:00:00")]
     forecast = IS.get_time_series(IS.DeterministicSingleTimeSeries, component, name)
     @test IS.get_interval(forecast) == Dates.Second(0)
 end
@@ -563,7 +565,8 @@ end
         interval,
     )
 
-    initial_times = IS.get_forecast_initial_times(sys)
+    initial_times = collect(IS.get_forecast_initial_times(sys))
+    @test initial_times == [Dates.DateTime("2020-01-01T00:00:00")]
     forecast = IS.get_time_series(IS.DeterministicSingleTimeSeries, component, name)
     @test IS.get_interval(forecast) == interval
 end

